akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala

```diff
@@ -105,24 +106,13 @@ trait LowPriorityToResponseMarshallerImplicits {
           val availableMarshallingsPerElement = source.mapAsync(1) { t ⇒ m(t)(ec) }
 
           // TODO optimise such that we pick the optimal marshalling only once (headAndTail needed?)
-          // TODO, NOTE: this is somewhat duplicated from Marshal.scala it could be made DRYer
           val bestMarshallingPerElement = availableMarshallingsPerElement map { marshallings ⇒
             // pick the Marshalling that matches our EntityStreamingSupport
-            val selectedMarshalling = s.contentType match {
-              case best @ (_: ContentType.Binary | _: ContentType.WithFixedCharset | _: ContentType.WithMissingCharset) ⇒
-                marshallings collectFirst { case Marshalling.WithFixedContentType(`best`, marshal) ⇒ marshal }
-
-              case best @ ContentType.WithCharset(bestMT, bestCS) ⇒
-                marshallings collectFirst {
-                  case Marshalling.WithFixedContentType(`best`, marshal) ⇒ marshal
-                  case Marshalling.WithOpenCharset(`bestMT`, marshal)    ⇒ () ⇒ marshal(bestCS)
-                }
-            }
-
-            // TODO we could either special case for certrain known types,
+            // TODO we could either special case for certain known types,
             // or extend the entity support to be more advanced such that it would negotiate the element content type it
             // is able to render.
-            selectedMarshalling.getOrElse(throw new NoStrictlyCompatibleElementMarshallingAvailableException[T](s.contentType, marshallings))
+            selectMarshallingForContentType(marshallings, s.contentType)
```

Comment:
`Marshal.scala` also does:
```
  orElse {
    marshallings collectFirst { case Marshalling.Opaque(marshal) ⇒ marshal }
  }
```
(https://github.com/akka/akka-http/pull/2004/files#diff-05f3ff570222d55bae3d318c06467f23R65)

Is there any reason why that's not also done here?

Reply:
Looks like it'd make sense to do that here as well. @jrudolph WDYT?

Reply:
I agree, otherwise `Opaque` marshallers won't work here.

Reply:
Okay; I'll have to add a test for this as well, right?

Reply:
Would be good, indeed. Would also be fine to create a ticket to add the test later. This PR would still be an improvement.

Reply:
Okay; added tests that check that an opaque marshaller will be used if and only if a non-opaque marshaller isn't available.